core/server/master/src/main/java/alluxio/master/metrics/DefaultMetricsMaster.java

```diff
@@ -125,7 +132,20 @@ public void start(Boolean isLeader) throws IOException {
   }
 
   @Override
-  public void putWorkerMetrics(InstanceType instance, String hostname, List<Metric> metrics) {
-    mMetricsStore.putWorkerMetrics(instance, hostname, metrics);
+  public void clientHeartbeat(String clientId, String hostname, List<Metric> metrics) {
+    if (metrics.isEmpty()) {
+      return;
+    }
```

Comment:
Could we do this logic in `putClientMetrics`, since it should be able to handle an empty set?